### PR TITLE
Fix TTR misuse

### PR DESCRIPTION
**Pull request by SkyLucilfer**
TTR() should be used with strings, not directly on variables, as the Python parser in extract.py only extract text within "". 

Edit: If a translation string has been extracted somewhere previously by TTR(), TTRC() etc., then using TTR() of that string contained in a variable should work.

### CODE REVIEW

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -2718,7 +2718,7 @@ void SceneTreeDock::_update_create_root_dialog() {
 				if (l != String()) {
 					Button *button = memnew(Button);
 					favorite_nodes->add_child(button);
-					button->set_text(TTR(l));
```
Comment (SkyLucilfer) [inline editor/scene_tree_dock.cpp:2721]: I think this one is impossible to fetch the text to be translated? 
The lines are from a file.

File: editor/input_map_editor.cpp [hunk 2]
```diff
@@ -36,44 +36,44 @@
...
-	"Home, DualShock PS, Guide",
-	"Start, Nintendo +",
-	"Left Stick, DualShock L3, Xbox L/LS",
-	"Right Stick, DualShock R3, Xbox R/RS",
-	"Left Shoulder, DualShock L1, Xbox LB",
-	"Right Shoulder, DualShock R1, Xbox RB",
-	"D-Pad Up",
-	"D-Pad Down",
-	"D-Pad Left",
-	"D-Pad Right"
```
Comment (SkyLucilfer) [inline editor/input_map_editor.cpp:53]: I think this one could benefit from adding context to TTRC, e.g.
TTRC("Face Bottom, DualShock Cross, Xbox A, Nintendo B", "Button_Input_Description");

File: editor/input_map_editor.cpp [hunk 3]
```diff
@@ -36,44 +36,44 @@
...
+	TTRC("Face Bottom, DualShock Cross, Xbox A, Nintendo B"),
+	TTRC("Face Right, DualShock Circle, Xbox B, Nintendo A"),
+	TTRC("Face Left, DualShock Square, Xbox X, Nintendo Y"),
```
Comment (SkyLucilfer) [inline editor/input_map_editor.cpp:41]: I think this one could benefit from adding context to TTRC, e.g.
TTRC("Face Bottom, DualShock Cross, Xbox A, Nintendo B", "Button_Input_Description");

File: editor/scene_tree_dock.cpp [hunk 4]
```diff
@@ -2718,7 +2718,7 @@ void SceneTreeDock::_update_create_root_dialog() {
...
+					button->set_text(l);
```
Comment (SkyLucilfer) [inline editor/scene_tree_dock.cpp:2721]: I think this one is impossible to fetch the text to be translated?
The lines are from a file.

File: editor/input_map_editor.cpp (hunk 3 above)
Comment (Calinou) [reply editor/input_map_editor.cpp:41]: That reminds me: if we're going to add translation contexts to the editor's strings, we should settle on a convention. I'd go for PascalCase as that's what we use for theme items already.

Comment (SkyLucilfer) [reply editor/input_map_editor.cpp:41]: Good idea. It also makes it clear that contexts should normally be short and not be used as comments for translators.

File: editor/script_create_dialog.cpp [hunk 5]
```diff
@@ -603,7 +603,7 @@ void ScriptCreateDialog::_path_entered(const String &p_path) {
 }
 
 void ScriptCreateDialog::_msg_script_valid(bool valid, const String &p_msg) {
-	error_label->set_text("- " + TTR(p_msg));
+	error_label->set_text("- " + p_msg);
```
Comment (SkyLucilfer) [inline editor/script_create_dialog.cpp:606]: TTR() already used when _msg_script_valid() is called.

File: editor/plugins/node_3d_editor_plugin.cpp [hunk 6]
```diff
@@ -5407,7 +5407,7 @@ void Node3DEditor::_update_gizmos_menu() {
 		}
 		String plugin_name = gizmo_plugins_by_name[i]->get_name();
 		const int plugin_state = gizmo_plugins_by_name[i]->get_state();
-		gizmos_menu->add_multistate_item(TTR(plugin_name), 3, plugin_state, i);
+		gizmos_menu->add_multistate_item(plugin_name, 3, plugin_state, i);
```
Comment (SkyLucilfer) [inline editor/plugins/node_3d_editor_plugin.cpp:5410]: gizmo_plugins_by_name[i]->get_name() function already has TTR() inside.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**